pkg/util/proxyutil/reverse_proxy.go

```diff
@@ -77,6 +77,9 @@ func wrapDirector(d func(*http.Request)) func(req *http.Request) {
 		d(req)
 		PrepareProxyRequest(req)
 
+		// Set X-Grafana-Referer to correlate access logs to dashboards
+		req.Header.Set("X-Grafana-Referer", req.Header.Get("Referer"))
+
```

Comment:
It would be nice to move this logic and deletion of Origin and Referer headers into  the PrepareProxyRequest function since then it will apply to both datasource proxy and call resource requests:
https://github.com/grafana/grafana/blob/10c2bdea3eec6c143bf59e326eacf000cdbb78d8/pkg/api/plugin_resource.go#L100

Reply:
Thanks for the quick feedback.  Good idea.  I moved the logic into PrepareProxyRequest and added a few tests.